frontend/src/api/k8s/roleBindings.ts

```diff
@@ -46,26 +46,28 @@ export const generateRoleBindingServingRuntime = (
   return roleBindingObject;
 };
 
-export const generateRoleBindingProjectSharing = (
+export const generateRoleBindingPermissions = (
   namespace: string,
-  rbSubjectType: ProjectSharingRBType,
+  rbSubjectType: RoleBindingPermissionsRBType,
   rbSubjectName: string,
-  rbRoleRefType: ProjectSharingRoleType,
+  rbRoleRefType: RoleBindingPermissionsRoleType | string, //string because with MR this can include MR name
```

Comment:
btw, one enhancement we should add is `ownerReferences` for the rolebinding, we should either add it into the service or the role, if we delete the modelregistry we would want all the rolebindings to be gone

Reply:
@lucferbux sorry, i'm not sure what you mean exactly. looks like `ownerReference` is part of the metadata for `K8sResourceCommon` so I could add it to the metadata of the roleBindingObject?

what values would i use though?
```
    apiVersion: string;
    kind: string;
    name: string;
    uid: string;
    controller?: boolean;
    blockOwnerDeletion?: boolean;
```

Reply:
I mean [garbage collection](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#owners-dependents). TBH I thought rolebindings were dependant of roles so when you deleted a role the rolebinding was deleted (not sure why i thought that was the behavior). But testing in the cluster, when I delete a model registry, all resoruces are deleted execpt rolebindings. Steps to replicate are:

1. Create a model registry in the admin panel
2. Manage access and add some users/groups
3. Delete the model registry
4. Check rolebindings in the namespace, rolebindings are still there

Just add a [owner reference](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications) to the role.

Reply:
We have a util method to add owner references [here](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/api/k8sUtils.ts#L7) let me know if you need anything.

Reply:
@lucferbux - i tried in [this commit](https://github.com/opendatahub-io/odh-dashboard/pull/2969/commits/dad2e6c1d3d4235da0c3f3bab1242674215caa25), but it doesn't seem to clear the rolebindings with the mr

Reply:
@lucferbux - i ended up using the MR CR, which required more changes to retrieve it and pass it through... but it seems to be working now

Reply:
Great, MR CR can be the owner reference since that's the higher order parent, so that's perfect.